src/cryptonote_protocol/cryptonote_protocol_handler.h

```diff
@@ -133,9 +135,9 @@ namespace cryptonote
     nodetool::i_p2p_endpoint<connection_context>* m_p2p;
     std::atomic<uint32_t> m_syncronized_connections_count;
     std::atomic<bool> m_synchronized;
-    bool m_one_request = true;
     std::atomic<bool> m_stopping;
-    epee::critical_section m_sync_lock;
+    std::mutex m_sync_lock;
```

Comment:
The last version of mingw I checked does not support `std::thread` or `std::mutex`.